App/Redux/PhotoViewingSelectors.ts

```diff
@@ -161,6 +161,23 @@ export function getThreadThumbs(state: RootState, byAddres: string, sortBy?: 'na
     })
 }
 
+function isDirectMessageThread(address: string): (thread: ThreadData) => boolean {
+  return (thread) => {
+    return thread.whitelist.indexOf(address) !== -1 &&
+           thread.sharing === Thread.Sharing.NOT_SHARED &&
+           thread.type === Thread.Type.OPEN
+  }
+}
+
+export function directMessageThreadExists(state: RootState, address: string): boolean {
+  return getThreads(state).some(isDirectMessageThread(address))
+}
+
+// Returns the thread object or undefined if it isn't found
+export function getDirectMessageThread(state: RootState, address: string): ThreadData | undefined {
```

Comment:
It's ok (even better) to just allow TypeScript infer the return types in all these functions.